src/test/ui/feature-gate/feature-gate-arbitrary_enum_discriminant.rs

```diff
@@ -0,0 +1,10 @@
+#![crate_type="lib"]
```

Comment:
and this... should also be moved into `ui/enum-discriminant/` and then use `// gate-test-arbitrary_enum_discriminant` in this specific file.

Reply:
Shall I move `test/ui/parser/tag-variant-disr-non-nullary.rs` too?

Reply:
I think that one can remain as-is since the change there is mostly transitory and will be noticed when stabilization is done anyways.

Reply:
> and this... should also be moved into `ui/enum-discriminant/` and then use `// gate-test-arbitrary_enum_discriminant` in this specific file.

The filename is already prefixed with `feature-gate-`. Is there a reason to prefer one convention for marking a file as a gate test over the other?

Reply:
I assumed tidy wouldn't be happy otherwise; other than that idk.